### PR TITLE
Purpose: To clean addresses for linkage work

### PR DESCRIPTION
**Pull request by tomc-hackney**
Python code that cleans and standardises full address data into a format optimised for record matching

### CODE REVIEW

**Review by LBH-wgreeff — APPROVED** [review]
LGTM

File: scripts/jobs/data_and_insight/address_cleaning.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+        for reg in inputs.clean_trim_address_regex_subs:
+            source_df = source_df.withColumn("clean_trim_address",
+                                             F.regexp_replace(F.col("clean_trim_address"), reg[0], reg[1]))
+
+        # tidy up extra spaces
+        source_df = source_df.withColumn("clean_trim_address",
+                                         F.regexp_replace(F.col("clean_trim_address"), " ?- ?$", ""))
+        source_df = source_df.withColumn("clean_trim_address", F.trim(F.col("clean_trim_address")))
+
```
Comment (LBHSBALLEY) [inline scripts/jobs/data_and_insight/address_cleaning.py:119]: Lines 116-118 would make a nice function

File: scripts/helpers/address_cleaning_inputs.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+"""Subs list for address cleaning"""
+
```
Comment (LBHSBALLEY) [inline scripts/helpers/address_cleaning_inputs.py:2]: In all the regex I would use "\s" instead of " " because it also covers white tabs

**Review by LBHSBALLEY — COMMENTED** [review]
Looks good but could use more regex to make the script shorter

**Pull request merged**